src/components/HealthCardDisplay/HealthCardDisplay.js

```diff
@@ -111,126 +112,75 @@ const HealthCardDisplay = () => {
   const history = useHistory();
   const { t } = useTranslation();
   const {
-    qrCodes, healthCardVerified, issuerVerified, validPrimarySeries
+    qrError,
+    jws,
   } = useQrDataContext();
-  const patientData = getPatientData(qrCodes);
 
   const handleScan = () => {
     history.push('qr-scan');
```

Comment:
```suggestion
    resetQrCodes();
    history.push('qr-scan');
```

Reply:
I ran into the following bug: if I scanned a QR that was not a valid SHC (stored UNSUPPORTED_QR_NOT_SHC for the `qrError`), then tried to scan a valid SHC right after, I would reach the invalid SHC error page for the valid SHC. It looks like the `qrError` doesn't get reset between scans, and `resetQrCodes()` never gets called. Adding `resetQrCodes()` to `handleScan()` solved the issue for me.

Reply:
The problem there is we don't want to clear the QR data until the scan is fully complete - if the person goes to qr-scan and then aborts, there is no data in the provider. If they navigate to the display page then it breaks. We want the display route to show the last scanned data stored in history, and resetQrCodes removes that info.

Reply:
Also, I wasn't able to replicate this error - tried scanning an unsupported card then a valid one and it successfully showed the valid one. Can you test again? And attach the QR codes you used for testing?

Reply:
When I tested this, I used a non-SHC QR (like [this one](https://en.wikipedia.org/wiki/QR_code#/media/File:QR_code_for_mobile_English_Wikipedia.svg) ) and then scanned the first QR from [our Confluence page](https://confluence.mitre.org/pages/viewpage.action?spaceKey=DVCI&title=QR+Codes+for+Testing+the+Web+Verifier). When scanning the second QR, I was brought to the error screen again.

Reply:
Got it, thanks for clarifying. I added the call to resetQrCodes() but at the point where the QR is actually done scanning, so we don't lose the old value in local storage if the user aborts during the scan. Tested with the codes you used and it seems to be working (fingers crossed).

Reply:
Sounds good, I tested with your changes and I don't run into the bug anymore.